src/main/java/betterquesting/api2/utils/QuestTranslation.java

```diff
@@ -23,17 +24,22 @@ public class QuestTranslation {
     /**
      * We'll look up translation keys directly from the map, to avoid needing to perform string
      * comparison to check if a key is missing.
+     *
+     * <p>This must be a supplier, because this class is called on the server (in order to build
+     * quest translation keys during saving). The {@link I18n} class is not available on the server,
+     * so attempting to fetch this map will cause a crash.
      */
-    private static final Map<String, String> translations;
-    static {
-        try {
-            Field localeField = ReflectionHelper.findField(I18n.class, "i18nLocale", "field_135054_a");
-            Field translationsField = ReflectionHelper.findField(Locale.class, "field_135032_a");
-            translations = (Map<String, String>) translationsField.get(localeField.get(null));
-        } catch (IllegalAccessException e) {
-            throw new RuntimeException(e);
-        }
-    }
+    private static final Supplier<Map<String, String>> translations =
+            Suppliers.memoize(
+                    () -> {
+                        try {
+                            Field localeField = ReflectionHelper.findField(I18n.class, "i18nLocale", "field_135054_a");
+                            Field translationsField = ReflectionHelper.findField(Locale.class, "field_135032_a");
```

Comment:
StatCollector provides i18n functions on both the client and server, would it have the necessary fields?

Reply:
Thanks; this is much better!